src/Microsoft.Azure.Batch.SoftwareEntitlement.Server/Controllers/SoftwareEntitlementsController.cs

```diff
@@ -57,43 +57,27 @@ public SoftwareEntitlementsController(ServerOptions serverOptions, ILogger logge
         [HttpPost]
         [Produces("application/json")]
         public IActionResult RequestEntitlement(
-            [FromBody] SoftwareEntitlementRequest entitlementRequest,
+            [FromBody] SoftwareEntitlementRequestBody entitlementRequestBody,
             [FromQuery(Name = "api-version")] string apiVersion)
         {
             try
             {
-                if (!IsValidApiVersion(apiVersion))
+                var (parameters, requestError) = TryExtractParameters(apiVersion, entitlementRequestBody);
+                if (requestError != null)
                 {
-                    return CreateInvalidApiVersionError(entitlementRequest, apiVersion);
+                    return CreateBadRequestResponse(requestError);
                 }
 
```

Comment:
It is checked first, but perhaps it's not obvious because it's in the same method with all the other validation. Perhaps we're looking from different viewpoints: I'm seeing the API version check as just one piece of validation (albeit an important one that occurs right at the start), but you're seeing it as something separate that occurs before the other validation starts?

Reply:
Different `apiVersion` values might trigger completely different code paths, depending on how the service evolves. I think we should keep the `apiVersion` check directly in the controller method so that it's an obvious pivot point.

Reply:
Fair enough, I can move it back.